migrations/versions/0398_non_nullable_international.py

```diff
@@ -0,0 +1,20 @@
+"""
+
+Revision ID: 0398_non_nullable_international
+Revises: 0397_missed_steps
+Create Date: 2023-01-19 16:09:03.772450
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+revision = "0398_non_nullable_international"
+down_revision = "0397_missed_steps"
+
+
+def upgrade():
+    op.alter_column("notifications", "international", existing_type=sa.BOOLEAN(), nullable=False)
```

Comment:
https://www.postgresql.org/docs/current/sql-altertable.html

> SET NOT NULL may only be applied to a column provided none of the records in the table contain a NULL value for the column. Ordinarily this is checked during the ALTER TABLE by scanning the entire table; however, if a valid CHECK constraint is found which proves no NULL can exist, then the table scan is skipped.

I'm worried that, as we don't have a matching check constraint, this will require alembic to scan the entire table to check for existance of null values.

Postgres acquires an `ACCESS EXCLUSIVE` lock for most alter statements, including this one. I'd definitely like us to ensure we can avoid any table scans while holding such a strict lock

side note: does alembic run each migration script in a separate transaction with commits between? I assume it does.

Reply:
Humm, yep, that one might be trickier.

re: side note - Alembic will run all of the (new) migrations in a single transaction: https://github.com/alphagov/notifications-api/blob/main/migrations/env.py#L64-L65

Reply:
Will take this migration out for now. We should be able to do something by adding an invalid check constraint, triggering a manual validation, and then doing the migration, but it's faffy...

Reply:
good to know about the single transaction 👍.

but yeah, i agree with solving it in a separate PR. we have patterns in past migrations for adding a constraint with the `not valid` flag, then calling `validate constraint` separately which acquires a much weaker lock